lib/phoenix_live_view/engine.ex

```diff
@@ -638,21 +633,42 @@ defmodule Phoenix.LiveView.Engine do
 
   defp changed_assigns(assigns) do
     checks =
-      for {key, _} <- assigns, not nested_and_parent_is_checked?(key, assigns) do
-        case key do
-          [assign] ->
-            quote do
-              unquote(__MODULE__).changed_assign?(changed, unquote(assign))
+      for {{changed_var, key}, _} <- assigns, not nested_and_parent_is_checked?(key, assigns) do
+        case changed_var do
+          :changed ->
+            case key do
+              [assign] ->
+                quote do
+                  unquote(__MODULE__).changed_assign?(changed, unquote(assign))
+                end
+
+              [assign | tail] ->
+                quote do
+                  unquote(__MODULE__).nested_changed_assign?(
+                    unquote(tail),
+                    unquote(assign),
+                    unquote(@assigns_var),
+                    changed
+                  )
+                end
             end
 
-          [assign | tail] ->
-            quote do
-              unquote(__MODULE__).nested_changed_assign?(
-                unquote(tail),
-                unquote(assign),
-                unquote(@assigns_var),
-                changed
-              )
+          :vars_changed ->
+            case key do
+              [assign] ->
+                quote do
+                  unquote(__MODULE__).changed_assign?(var!(vars_changed), unquote(assign))
+                end
+
+              [assign | tail] ->
+                quote do
+                  unquote(__MODULE__).nested_changed_assign?(
+                    unquote(tail),
+                    unquote(assign),
+                    unquote(@assigns_var),
+                    var!(vars_changed)
+                  )
+                end
```

Comment:
this can probably be more concise, but I failed to construct the correct variable, since one requires `var!` and one does not